dev-lang/rust/rust-1.46.0.ebuild

```diff
@@ -198,6 +196,31 @@ src_prepare() {
 
 		"${WORKDIR}/${rust_stage0}"/install.sh --disable-ldconfig \
 			--destdir="${rust_stage0_root}" --prefix=/ || die
+
+		if use prefix; then
+			interpreter=`ldconfig -p | grep ld-linux | awk '{print $NF}' || die "Cannot find your ld.so, why?"`
+			ebegin "Changing interpreter to $interpreter for Gentoo prefix"
+			for f in `ls -Ud ${rust_stage0_root}/bin/*  || die "Cannot find binary of rust_stage0, why?"`; do
```

Comment:
don't use ls as input, use `find -exec` instead

also make variables local. and ditto for backticks.